src/main/java/uk/gov/hmcts/darts/dailylist/service/DailyListProcessorImpl.java

```diff
@@ -0,0 +1,187 @@
+        for (DailyListEntity dailyList : dailyLists) {
+            dailyList.setStatus(String.valueOf(JobStatusType.IGNORED));
+        }
+    }
+
+    @SuppressWarnings("checkstyle:VariableDeclarationUsageDistance")
+    private void processDailyList(DailyListEntity dailyListEntity) throws JsonProcessingException {
+        DailyList dailyList = objectMapper.readValue(dailyListEntity.getContent(), DailyList.class);
+        JobStatusType statusType = JobStatusType.PROCESSED;
+
+        for (CourtList courtList : dailyList.getCourtLists()) {
+
+            String courtHouseName = courtList.getCourtHouse().getCourtHouseName();
+            Optional<CourthouseEntity> foundCourthouse = courthouseRepository.findByCourthouseNameIgnoreCase(
+                    courtHouseName);
+
+            if (foundCourthouse.isPresent()) {
+                List<Sitting> sittings = courtList.getSittings();
+                for (Sitting sitting : sittings) {
+                    List<Hearing> hearings = sitting.getHearings();
+                    for (Hearing dailyListHearing : hearings) {
+                        CourtCaseEntity courtCase = retrieveCoreObjectService.retrieveOrCreateCase(
+                                courtHouseName,
+                                dailyListHearing.getCaseNumber()
+                        );
+                        HearingEntity hearing = retrieveCoreObjectService.retrieveOrCreateHearing(
+                                courtHouseName,
+                                String.valueOf(sitting.getCourtRoomNumber()),
+                                dailyListHearing.getCaseNumber(),
+                                dailyListHearing.getHearingDetails().getHearingDate()
+                        );
+
+                        addJudges(sitting, hearing);
+                        addDefendants(courtCase, dailyListHearing.getDefendants());
+                        addProsecutions(
+                                courtCase,
+                                buildFullName(dailyListHearing.getProsecution().getAdvocate().getPersonalDetails().getName())
+                        );
+                        addDefenders(courtCase, dailyListHearing.getDefendants());
+                        setCourtroom(hearing, courtCase, sitting.getCourtRoomNumber());
+                        setCaseNumber(courtCase, dailyListHearing.getDefendants(), dailyListEntity, dailyListHearing);
+                        hearing.setHearingDate(dailyListHearing.getHearingDetails().getHearingDate());
+
+                        courtCase.addHearing(hearing);
+                        hearingRepository.saveAndFlush(hearing);
+                        caseRepository.saveAndFlush(courtCase);
+
+
+                    }
+                }
+            } else {
+                statusType = JobStatusType.PARTIALLY_PROCESSED;
+                log.info("Unregistered courthouse " + courtHouseName + " daily list entry has not been processed");
```

Comment:
should probably be ERROR instead of INFO

Reply:
ACCEPT